### DISCUSSION TITLE
Credentials for two (or more) private repos on one host?

### DISCUSSION THREAD

**Discussion by jstandard** [sbt/sbt#7584] [Q&A] [answered]
Hello,  

I'm attempting to set up my CI/CD environment.  The build needs to access two private repos on one Artifactory instance.  When I perform the build locally, I can just pull the credentials from my local `~/.sbt/.credentials` file since I have access to both private repos and any public artifacts on that host.  However, when the gitlab-runner runs, I want to use encrypted environment variables for each of these private repos. Something like...

```
credentials += {
  List("USERNAME_ONE", "PASSWORD_ONE").flatMap(sys.env.get) match {
    case List(user, password) =>
      streams.value.log.info(s"private/internal dependencies resolution credentials for host host.name.org user '$user' has been configured")
      Some(Credentials("Artifactory Realm", "host.name.org", user, password))
    case _ =>
      streams.value.log.warn(
        "private/internal dependencies resolution may fail as some environment variables are missing : USERNAME_ONE / PASSWORD_ONE")
      None
  }
}

credentials += {
  List("USERNAME_TWO", "PASSWORD_TWO").flatMap(sys.env.get) match {
    case List(user, password) =>
      streams.value.log.info(s"private/internal dependencies resolution credentials for host host.name.org user '$user' has been configured")
      Some(Credentials("Artifactory Realm", "host.name.org", user, password))
    case _ =>
      streams.value.log.warn(
        "private/internal dependencies resolution may fail as some environment variables are missing : USERNAME_TWO / PASSWORD_TWO")
      None
  }
}
```
Does this look right?  I haven't been able to get this to work.  Since we are using the same host for both repositories, should I use host plus the <repo-key> (i.e., path to private repo) in the Credentials interface?  For instance...

`Credentials("Artifactory Realm", "host.name.org"+"/path/to/private/repo", user, password)`

**Comment by e-orz** [comment, marked as answer]
Hi,
We had the same issue in our company, so I wrote a sbt plugin to handle it. It just maps between a resolver to its credentials. You can create two resolvers and map each one to its specific credentials.
see:
https://github.com/e-orz/sbt-resolver-credentials

**Reply by jstandard** [reply]
Thanks!  I just got a chance to use your plugin.  It worked as described...

**Reply by hertg** [reply]
Unfortunately, this doesn't work in my project that I'm building and publishing via GitHub Actions.

I have a library that uses dependencies published as **private** packages to GitHub Package Registry. This library should also be published itself to GitHub Package Registry. I want to use an organization secret ("MAVEN_READ_PAT") to download dependencies, but use the GitHub Action provided `GITHUB_TOKEN` to publish the library. This way, the read-only token would suffice as organization secret, and the write token to be used for uploading is always scoped to this repository. But I'm giving up trying to achieve this, I've lost too much time trying to get this to work.
